# HG changeset patch
# User Juan Pablo Carbajal <ajuanpi+dev@gmail.com>
# Date 1536367178 -7200
#      Sat Sep 08 02:39:38 2018 +0200
# Branch stable
# Node ID 6e41caf35af76527f792b8b62583a244bd355732
# Parent  778a0400586b6587c4c36e690712474c20ec5abe
doc: unmkpp.m same words as mkpp, add example, explain coefficients (bug #54523)

diff --git a/scripts/polynomial/unmkpp.m b/scripts/polynomial/unmkpp.m
--- a/scripts/polynomial/unmkpp.m
+++ b/scripts/polynomial/unmkpp.m
@@ -16,31 +16,49 @@
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {[@var{x}, @var{p}, @var{n}, @var{k}, @var{d}] =} unmkpp (@var{pp})
 ##
 ## Extract the components of a piecewise polynomial structure @var{pp}.
 ##
+## This function is the inverse of @code{mkpp}: it extracts the inputs to
+## @code{mkpp} needed to create the piecewise polynomial structure @var{PP}.
+## The code below makes this relation explicit:
+##
+## @example
+## @group
+## [breaks, coefs, numinter, order, dim] = unmkpp (pp);
+## pp2  = mkpp (breaks, coefs, dim);
+## @end group
+## @end example
+##
+## The piecewise polynomial structure @code{pp2} obtained in this way, is
+## identical to the original @code{pp}.
+## The same can be obtained by directly accessing the fields of the structure
+## @code{pp}.
+##
 ## The components are:
 ##
 ## @table @asis
 ## @item @var{x}
-## Sample points.
+## Sample points or breaks.
 ##
 ## @item @var{p}
 ## Polynomial coefficients for points in sample interval.
-## @code{@var{p} (@var{i}, :)} contains the coefficients for the polynomial
-## over interval @var{i} ordered from highest to lowest.  If
-## @code{@var{d} > 1}, @code{@var{p} (@var{r}, @var{i}, :)} contains the
-## coefficients for the r-th polynomial defined on interval @var{i}.
+## @code{@var{p}(@var{i}, :)} contains the coefficients for the polynomial
+## over interval @var{i} ordered from highest to lowest degree.
+## If @code{@var{d} > 1}, then @var{p} is a matrix of size
+## @code{[@var{n}*prod(@var{d}) @var{m}]}, where the @code{@var{i} + (1:@var{d})} rows are the
+## coefficients of all the @var{d} polynomials in the interval @var{i}.
 ##
 ## @item @var{n}
-## Number of polynomial pieces.
+## Number of polynomial pieces or intervals,
+## @code{@var{n} = length (@var{x}) - 1}.
 ##
 ## @item @var{k}
 ## Order of the polynomial plus 1.
 ##
 ## @item @var{d}
 ## Number of polynomials defined for each interval.
 ## @end table
 ##
